src/adapter/graphql/src/queries/datasets/dataset_metadata.rs

```diff
@@ -79,77 +84,188 @@ impl DatasetMetadata {
         }
     }
 
+    // TODO: Private Datasets: tests
     /// Current upstream dependencies of a dataset
-    async fn current_upstream_dependencies(&self, ctx: &Context<'_>) -> Result<Vec<Dataset>> {
-        let (dependency_graph_service, dataset_registry) = from_catalog_n!(
+    async fn current_upstream_dependencies(
+        &self,
+        ctx: &Context<'_>,
+    ) -> Result<Vec<DependencyDatasetResult>> {
+        let (
+            dependency_graph_service,
+            dataset_action_authorizer,
+            dataset_entry_repository,
+            account_service,
+        ) = from_catalog_n!(
             ctx,
             dyn domain::DependencyGraphService,
-            dyn domain::DatasetRegistry
+            dyn kamu_core::auth::DatasetActionAuthorizer,
```

Comment:
I'm not a fan of using authorizer in GraphQL too. As we discussed, we should have some r/o use cases as well, and authorizers naturally fit there.

The handlers for dependencies , upstream and downstream, seem overcomplicated. This is an API entry, we clearly want 90% of this logic in a use case. The handler should make a call and convert the results into GraphQL-compatible outputs, that's it. 

Imagine we will need a REST API equivalent of these endpoints...

Reply:
Extracted to two use-cases:
- `GetDatasetDownstreamDependenciesUseCase`
- `GetDatasetUpstreamDependenciesUseCase`